source/air/src/main.rs

```diff
@@ -126,6 +126,10 @@ pub fn main() {
             ValidityResult::TypeError(err) => {
                 panic!("Type error: {}", err);
             }
+            ValidityResult::SingularInvalid(..) => {
+                count_errors += 1;
+                println!("SingularInvalid in air main - Singular error at: TODO");
```

Comment:
There's a leftover TODO here, which I imagine is turning this into proper error reporting instead of a println.